lib/client/doc.js

```diff
@@ -842,8 +843,8 @@ Doc.prototype._opAcknowledged = function(message) {
     // have sent all the ops that have happened before acknowledging our op
     logger.warn('Invalid version from server. Expected: ' + this.version + ' Received: ' + message.v, message);
 
-    // Fetching should get us back to a working document state
-    return this.fetch();
+    // Rollback to clear the inflightOp and trigger a fetch from the server
+    return this._rollback();
```

Comment:
I suspect that the reason this was an issue for you is that the server was not keeping versions canonical. I suspect that this will not be needed once you make changes to the server that keeps version numbers canonical and increasing by 1 for each operation.

We should avoid doing a rollback unless completely necessary as it is a method of last resort to get the document back into working order. The downside to rollback is that it will clear any local state, which means we will lose data.